pyproj/_geod.pyx

```diff
@@ -90,10 +90,7 @@ cdef class Geod:
         geod_init(&self._geod_geodesic, <double> a, <double> f)
         self.a = a
         self.f = f
-        # convert 'a' only for initstring
-        a_str = int(a) if a.is_integer() else a
-        f_str = int(f) if f.is_integer() else f
-        self.initstring = f"+a={a_str} +f={f_str}"
+        self.initstring = f"+{a=} +{f=}"
```

Comment:
Note that I took the easy route and changed the repr to a pure float for `Geod`. Does this matter? The actual usage is as a double, so presumably this is actually "more correct" to show to users.

Reply:
This change predates me. Is it relevant to this PR?

Reply:
Yes, there is a borrowed Python reference error with it so the project won't even compile. We could change this to something like setting a temporary variable and expanding the if/else.

Reply:
I think that your fix makes the most sense to me.